onnxruntime/python/tools/quantization/fp16_converter.py

```diff
@@ -0,0 +1,359 @@
+import argparse
+import itertools
+
+import numpy as np
+import onnx
+import packaging.version as pv
+from onnx import helper, numpy_helper
+from onnx import onnx_pb as onnx_proto
+
+from onnxruntime.quantization.onnx_model import ONNXModel
+
+# from onnxruntime.transformers.float16 import convert_tensor_float_to_float16
+
+ALLOWED_OPS_LIST = ["Conv", "MatMul"]
+
+
+def make_value_info_from_tensor(tensor):
+    shape = numpy_helper.to_array(tensor).shape
+    return helper.make_tensor_value_info(tensor.name, tensor.data_type, shape)
+
+
+def _npfloat16_to_int(np_list):
+    """
+    Convert numpy float16 to python int.
+
+    :param np_list: numpy float16 list
+    :return int_list: python int list
+    """
+    return [int(bin(_.view("H"))[2:].zfill(16), 2) for _ in np_list]
+
+
+def convert_np_to_float16(np_array, min_positive_val=5.96e-08, max_finite_val=65504.0):
+    """
+    Convert float32 numpy array to float16 without changing sign or finiteness.
+    Positive values less than min_positive_val are mapped to min_positive_val.
+    Positive finite values greater than max_finite_val are mapped to max_finite_val.
+    Similar for negative values. NaN, 0, inf, and -inf are unchanged.
+    """
+
+    def between(a, b, c):
+        return np.logical_and(a < b, b < c)
+
+    np_array = np.where(between(0, np_array, min_positive_val), min_positive_val, np_array)
```

Comment:
I am not sure but is performance a concern here? it scans the array many times.

Reply:
Copied from float16.py. Performance is not the main concerns here I believed.